app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedFragment.kt

```diff
@@ -26,17 +30,27 @@ class FeedFragment : Fragment() {
     private lateinit var viewModel: FeedViewModel
     private lateinit var feedAdapter: FeedAdapter
 
-    override fun onCreateView(inflater: LayoutInflater,
-                              container: ViewGroup?,
-                              savedInstanceState: Bundle?): View? {
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
         dataBinding = DataBindingUtil.inflate(inflater, R.layout.feed_fragment, container, false)
         setupViewModel()
         setupViews()
         return dataBinding.root
     }
 
     private fun setupViewModel() {
-        viewModel = ViewModelProvider(this, FeedViewModelFactory())
+
+        val repository = FeedRepository(
+            FeedApi, Room.databaseBuilder(
+                this.context!!, // TODO understant !! operator
+                FeedDatabase::class.java, "feed-database"
+            ).build()
+        )
```

Comment:
Instead of creating a new instance of `FeedDatabase` and `FeedRepository` in every `Fragment` (and our app could have many of them) and also each time `FeedFragment` is created (for example when a user turns the phone from portrait to landscape and the other way) we would like to use a single instance of both classes.

Here's what you can do in FeedDatabase.kt:
```
private lateinit var INSTANCE: FeedDatabase

fun getDatabase(context: Context): FeedDatabase {
    synchronized(FeedDatabase::class.java) {
        if (!::INSTANCE.isInitialized) {
            INSTANCE = ...
        }
    }

    return INSTANCE
}
```

Reply:
I also suggest to move the creation of `FeedRepository` to `FeedViewModelFactory`.

Reply:
Instead of `this.context!!` you can write `requireContext()`.